app/controllers/posts_controller.rb

```diff
@@ -22,6 +36,42 @@ def create
     end
   end
 
+  def update
+    # remove blank tag value
+    # convert from tag_id to tag object
+    values = post_params
+    values[:tags].shift
+    values[:tags].map! { |tag| Tag.find(tag) }
+
+    # set cover_image nil if no image is attached
+    @post.cover_image.purge if values[:remove_cover_image] == '1' && !values.key?(:cover_image)
+    values.delete(:remove_cover_image)
+
+    if @post.update(values)
+      redirect_to @post
+    else
+      render 'edit'
+    end
+  end
+
+  def accept
+    @post.update(status: :accepted)
+
+    redirect_to @post
+  end
+
+  def reject
+    @post.update(status: :rejected)
+
+    redirect_to @post
+  end
+
+  def destroy
+    @post.destroy
+
+    redirect_to posts_path
+  end
```

Comment:
Assume that all of these actions are just belongs to admin user, you can write a callback to secure the action for example:

```
before_action :require_admin, only: %i[accept reject destroy]


private

  def require_admin
    redirect_to root_path unless current_user.admin?
  end
```